src/test/tfs_rest_parser_binary_inputs_test.cpp

```diff
@@ -98,6 +98,35 @@ TEST_F(TFSRestParserBinaryInputs, RowString) {
     EXPECT_EQ(strcmp(parser.getProto().inputs().find("i")->second.string_val(0).c_str(), "abcd"), 0);
 }
 
+TEST_F(TFSRestParserBinaryInputs, RowStringInvalidPrecision) {
+    std::string request = R"({"signature_name":"","instances":[{"i":"abcd"}]})";
+
+    TFSRestParser parser(prepareTensors({{"i", {-1, -1}}}, ovms ::Precision::FP32));
+    ASSERT_EQ(parser.parse(request.c_str()), StatusCode::REST_COULD_NOT_PARSE_INSTANCE);
+}
+
+TEST_F(TFSRestParserBinaryInputs, RowStringInvalidShape) {
+    std::string request = R"({"signature_name":"","instances":[{"i":"abcd"}]})";
+
+    TFSRestParser parser(prepareTensors({{"i", {-1, -1, -1}}}, ovms ::Precision::U8));
+    ASSERT_EQ(parser.parse(request.c_str()), StatusCode::OK);
+    ASSERT_EQ(parser.getProto().inputs_size(), 1);
+    ASSERT_EQ(parser.getProto().inputs().count("i"), 1);
+    ASSERT_EQ(parser.getProto().inputs().find("i")->second.string_val_size(), 1);
+    EXPECT_EQ(strcmp(parser.getProto().inputs().find("i")->second.string_val(0).c_str(), "abcd"), 0);
```

Comment:
Please assert shape of such request

Reply:
In all the added tests assertions for proto shape and precisions are missing @bstrzele